components/automatic_data_extraction/actions/extract-data-from-url/extract-data-from-url.mjs

```diff
@@ -0,0 +1,69 @@
+import { clearObj } from "../../common/utils.mjs";
+
+export default {
+  key: "automatic_data_extraction-extract-data-from-url",
+  name: "Extract Data From URL",
+  version: "0.0.1",
+  description: "Initiates the process of extracting data from a specified URL [See the docs here](https://docs.zyte.com/automatic-extraction-integrations.html#node-js)",
+  type: "action",
+  props: {
+    url: {
+      type: "string",
+      label: "URL",
+      description: "URL of web page to extract from. Must be a valid `http://` or `https:// URL`.",
+    },
+    pageType: {
+      type: "string",
+      label: "Page Type",
+      description: "Type of extraction to perform.",
+      options: [
+        "article",
+        "articleList",
+        "comments",
+        "forumPosts",
+        "jobPosting",
+        "product",
+        "productList",
+        "realEstate",
+        "reviews",
+        "vehicle",
+      ],
+    },
+    meta: {
+      type: "string",
+      label: "Meta",
+      description: "User UTF-8 string, which will be passed through the extraction pipeline and returned in the query result. Max size 4 Kb.",
+    },
+    articleBodyRaw: {
+      type: "boolean",
+      label: "Article Body Raw",
+      description: "Whether or not to include article HTML in article extractions. True by default. Setting this to false can reduce response size significantly if HTML is not required.",
+    },
+    fullHtml: {
+      type: "boolean",
+      label: "Full HTML",
+      description: "Include the full, raw HTML of the target web page in the query result. This is a premium feature that is disabled by default. [Open a support ticket](https://support.zyte.com/support/tickets/new) if you wish to have it enabled for your account.",
+    },
+    customHtml: {
+      type: "string",
+      label: "Custom HTML",
+      description: "HTML source to be scraped. Extraction will be done from the provided HTML with additional resources (images, CSS, etc.) downloaded from the provided url. JavaScript processing will be disabled. The String should be UTF-8 encoded. The maximum length is 2,000,000 characters, longer requests will be rejected.",
+    },
+  },
+  async run({ $ }) {
+    const {
+      automaticDataExtraction,
+      ...data
+    } = this;
+
+    const response = await automaticDataExtraction.extractData({
+      $,
+      data: JSON.stringify([
+        clearObj(data),
+      ]),
+    });
+
+    $.export("$summary", "The process of extracting data was successfully extracted!");
```

Comment:
```suggestion
    $.export("$summary", "The data was successfully extracted!");
```

Reply:
The process actually extracts the data. I think you got confused because the task asks for a action that starts an asynchronous proccess, but it'ss not possible, the request already returns the extract data.

Reply:
Sorry again, I forgot to update the description. This action sends a URL to extract and the response is the extracted data